hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/DefaultPortableReader.java

```diff
@@ -684,8 +687,11 @@ private PortablePosition findPositionForReading(String path) throws IOException
         }
     }
 
-    @SuppressWarnings({"checkstyle:cyclomaticcomplexity", "checkstyle:returncount"})
+    @SuppressWarnings({"checkstyle:cyclomaticcomplexity", "checkstyle:returncount", "unchecked"})
     private <T> T readSinglePositionFromNonArray(PortablePosition position) throws IOException {
+        if (position.getType() == null) {
+            throw new IllegalArgumentException("Unsupported type read: null");
+        }
         switch (position.getType()) {
```

Comment:
Yes, just saying: If you don't trigger the asserted lines, JaCoCo will count this as missing branch coverage. So it's just like a normal `if` which needs a test with your mentioned `@RequireAssertEnabled` annotation.